### ISSUE TITLE
ScrollView's scrollToEnd scrolls too far on android

### ISSUE THREAD

**Issue by coerdes-chris** [facebook/react-native#47959]
### Description

ScrollView's `scrollToEnd` method gives different results on iOs and Android.

Expected behaviour:
ScrollView's `scrollToEnd` method should function consistently across platforms. Ideally The view should scroll until the bottom of the content reaches the bottom of the view.

Actual behaviour:
On iOs the view tries to scroll until the bottom of the content reaches the bottom of the view.
On Android the view tries to scroll until the bottom of the content reaches the top of the view.

This is especially annoying if you add content while the view is scrolling, as it then scrolls too far, or if you have an animation of content being added to slide in from the bottom.

### Steps to reproduce

1. Open the provided snack on Android
2. Press the button labeled "Press me!"
3. The view scrolls until element "Test 50" reaches the top of the screen
4. Open the provided snack on iOs
5. Press the button labeled "Press me!"
6. The view scrolls until element "Test 49" reaches the bottom of the screen

### React Native Version

0.76.3

### Affected Platforms

Runtime - Android

### Output of `npx react-native info`

```text
System:
  OS: macOS 14.7
  CPU: (10) arm64 Apple M1 Pro
  Memory: 124.23 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.11.1
    path: ~/.nvm/versions/node/v20.11.1/bin/node
  Yarn:
    version: 1.22.19
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.2.4
    path: ~/.nvm/versions/node/v20.11.1/bin/npm
  Watchman:
    version: 2024.09.16.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.1
      - iOS 18.1
      - macOS 15.1
      - tvOS 18.1
      - visionOS 2.1
      - watchOS 11.1
  Android SDK: Not Found
IDEs:
  Android Studio: 2024.2 AI-242.23339.11.2421.12550806
  Xcode:
    version: 16.1/16B40
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 18.0.2.1
    path: /Users/chriscoerdes/.jenv/shims/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.0.1
    wanted: 15.0.1
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.76.3
    wanted: 0.76.3
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
N/A
```

### Reproducer

https://github.com/coerdes-chris/react-native-scrollview

### Screenshots and Videos

https://github.com/user-attachments/assets/ad2a0bcd-4c59-4291-8891-9f2dfbd1d13f

**Comment by oddlyspaced** [comment]
I tried the reproducer code. For now adding a setTimeout for the operation that needs to be done immediately after scrolling seems to solve the issue on Android.

**Comment by coerdes-chris** [comment]
> I tried the reproducer code. For now adding a setTimeout for the operation that needs to be done immediately after scrolling seems to solve the issue on Android.

Although this does prevent the ScrollView from scrolling too far, it doesn't really solve my issue. We have a questionnaire in our app, where new questions depend on the answer of the previous question. New questions are supposed to 'slide in' from the bottom. The way I accomplish this currently, is by rendering the new question at the bottom of the ScrollView, but invisible, then calling `scrollToEnd()`, and then immediately moving the question down, making it visible, and sliding it up using react-native-reanimated. In your proposed workaround, I would have to render the question (invisible), call `scrollToEnd()`, wait for the scrolling to finish, and then animate the question 'sliding in' from the bottom. This would look quite weird.

**Comment by oddlyspaced** [comment]
Correct, that was just my finding. Ideally, it should be fixed in the source. 🙏
